### PR TITLE
Simplify archery integration and disable JS temporarily

### LINKED ISSUE #2448: Integration Test Failures
**Describe the bug**

Integration tests are failing with

```
gulp-google-closure-compiler: java.util.zip.ZipException: invalid entry CRC (expected 0x4e1f14a4 but got 0xb1e0eb5b)
```

**To Reproduce**

**Expected behavior**

Integration test should not fail

**Additional context**

This appears to be an issue with zlib - https://github.com/google/closure-compiler-npm/issues/234 

In particular https://github.com/madler/zlib/issues/613

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #2448

# Rationale for this change
 

The current integration test has the following flow

- Github CI orchestrates a python script
- Which orchestrates docker compose
- Which orchestrates docker
- Which runs scripts in the arrow repo
- Which runs different python scripts

This is not only immensely confusing, but makes reproducing issues very difficult and complicates opting out of certain integration tests, e.g. because they are broken #2448 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -46,22 +51,46 @@ jobs:
...
+      - name: Build C#
+        run: ci/scripts/csharp_build.sh . /build
+      - name: Build Go
+        run: ci/scripts/go_build.sh .
+      - name: Build Java
+        run: ci/scripts/java_build.sh . /build
+      - name: Build JS
+        run: ci/scripts/js_build.sh . /build
+      - name: Install archery
+        run: pip install -e dev/archery
```
Comment (tustvold) [inline .github/workflows/integration.yml:69]: Sourced from https://github.com/apache/arrow/blob/master/ci/scripts/integration_arrow.sh

**Review by alamb — COMMENTED** [review]
I am +0 on this change (I don't oppose it but I am not a fan either). I would like some other opinions

My hesitation with moving away from archery.py (which is complicated, as you note in the description) is that archery is used for testing the other implementations against each other, so by making this change we are effectively committing to maintaining the equivalent integration scripts (e.g. adding new images and/or keeping up to date with new dependencies or tests). 

Maybe this isn't a big deal, but I would prefer not to have to maintain it

cc @viirya

**Comment by tustvold** [comment]
> My hesitation with moving away from archery.py

This PR doesn't move away from using archery.py, the tests are still orchestrated by it. All it moves away from is the docker-compose plumbing within the arrow repo proper, which is just unnecessary

File: .github/workflows/integration.yml [hunk 2]
```diff
@@ -46,22 +69,47 @@ jobs:
...
+      - name: Build Java
+        run: conda run --no-capture-output ci/scripts/java_build.sh $PWD /build
+      # Temporarily disable JS https://issues.apache.org/jira/browse/ARROW-17410
+      # - name: Build JS
+      #   run: conda run --no-capture-output ci/scripts/js_build.sh $PWD /build
+      - name: Install archery
+        run: conda run --no-capture-output pip install -e dev/archery
+      - name: Run integration tests
+        run: |
+          conda run --no-capture-output archery integration \
```
Comment (viirya) [inline .github/workflows/integration.yml:91]: Do we need `conda run` for each commend? I think the docker image enables the conda environment by default?

File: .github/workflows/integration.yml [hunk 3]
```diff
@@ -46,22 +69,47 @@ jobs:
...
+            --with-java=1 \
+            --with-js=0 \
+            --with-go=1 \
+            --with-rust=1 \
+            --gold-dirs=testing/data/arrow-ipc-stream/integration/0.14.1 \
+            --gold-dirs=testing/data/arrow-ipc-stream/integration/0.17.1 \
+            --gold-dirs=testing/data/arrow-ipc-stream/integration/1.0.0-bigendian \
+            --gold-dirs=testing/data/arrow-ipc-stream/integration/1.0.0-littleendian \
+            --gold-dirs=testing/data/arrow-ipc-stream/integration/2.0.0-compression \
+            --gold-dirs=testing/data/arrow-ipc-stream/integration/4.0.0-shareddict
```
Comment (viirya) [inline .github/workflows/integration.yml:104]: I think the change here is to move executing commends from docker-compose (`conda-integration`) to separate CI steps. The advantage is easier to control each step for each implementations (e.g. disabling JS).

Although we don't move from `archery`, there is still a slight disadvantage that we maintain the execution here and if any change is made in arrow repo, we might miss it at first.

There is some overhead but not as much as moving from `archery`, I think.

File: .github/workflows/integration.yml (hunk 2 above)
Comment (tustvold) [reply .github/workflows/integration.yml:91]: Sadly github actions overrides the entrypoint, among other things, and so this is the only way to make it work..

File: .github/workflows/integration.yml (hunk 3 above)
Comment (tustvold) [reply .github/workflows/integration.yml:104]: Correct, this should be relatively straightforward to maintain, and we can always revert back should issues arise

**Review by viirya — APPROVED** [review]

File: .github/workflows/integration.yml (hunk 3 above)
Comment (viirya) [reply .github/workflows/integration.yml:104]: > Aside from giving more control, the CI reports will be more helpful as you can see the stage that failed, and it is significantly easier imo to understand what is going on and how to reproduce it locally.

Oh, yea, that's right, forgot this point.

**Comment by viirya** [comment]
This looks good to me. Alternatively, if we don't like this approach, we may need to update docker-compose command at arrow repo to disable JS as it doesn't take any environment variables for that.

**Comment by tustvold** [comment]
I think the JS issue has possibly been fixed upstream, but regardless the capability to selectively disable broken integration tests still feels valuable, not to mention the other advantages to reproducibility and intelligibility of this PR

I'll leave this PR open for now in case anyone else wishes to weigh in, otherwise I'll merge it tomorrow morning

**Comment by alamb** [comment]
I wonder when we will re-enable the javacript integration test?

**Pull request merged**